stylix/mk-target.nix

```
@@ -0,0 +1,219 @@
+/**
+  Provides a consistent target interface, minimizing boilerplate and
+  automatically safeguarding declarations related to disabled options.
+
+  # Type
+
+  ```
+  mkTarget :: AttrSet -> ModuleBody
+  ```
+
+  # Examples
+
+  This function returns the module body, not an actual module.
+
+  The `modules/«MODULE»/«PLATFORM».nix` modules should use this function as
+  follows:
+
+      { mkTarget, lib... }:
+      mkTarget {
+        name = "«name»";
+        humanName = "«human readable name»";
+
+        generalConfig =
+          lib.mkIf complexCondition {
+            home.packages = [ pkgs.hello ];
+          };
+
+        configElements = [
+          { programs.«name».theme.name = "stylix"; }
+
+          (
+            { colors }:
+            {
+              programs.«name».theme.background = colors.base00;
+            }
+          )
+
+          (
+            { fonts }:
+            {
+              programs.«name».font.name = fonts.monospace.name;
+            }
+          )
+        ];
+      }
+
+  Note that `mkTarget` is provided in the same place as module arguments.
+
+  Ideally, in the future, this function returns an actual module by better
+  integrating with the /stylix/autoload.nix logic, allowing the following target
+  simplification and preventing access to unguarded module arguments by
+  requiring /modules/<MODULE>/<PLATFORM>.nix files to be attribute sets instead
+  of modules:
+
+      {
+        name = "«name»";
+        humanName = "«human readable name»";
+
+        generalConfig =
+          { lib, pkgs }:
+          lib.mkIf complexCondition {
+            home.packages = [ pkgs.hello ];
+          };
+
+        configElements = [
+          { programs.«name».theme.name = "stylix"; }
+
+          (
+            { colors }:
+            {
+              programs.«name».theme.background = colors.base00;
+            }
+          )
+
+          (
+            { fonts }:
+            {
+              programs.«name».font.name = fonts.monospace.name;
+            }
+          )
+        ];
+      }
+
+  # Inputs
+
+  : `name`
+    : The target name used to generate options in the stylix.targets.${name}
+      namespace.
+
+  : `humanName`
+    : The descriptive target name passed to the lib.mkEnableOption function
+      when generating the stylix.targets.${name}.enable option.
+
+  : `autoEnable`
+    : Whether the target should be automatically enabled by default according
+      to the stylix.autoEnable option.
+
+      This should be disabled if manual setup is required or if auto-enabling\
+      causes issues.
+
+  : `extraOptions`
+    : Additional options to be added in the stylix.targets.${name} namespace
+      along the stylix.targets.${name}.enable option.
+
+      For example, an extension guard used in the configuration can be declared
+      as follows:
+      { extension.enable = lib.mkEnableOption "the bloated dependency"; }
```

If these blocks are plain text, this should be indented:

```suggestion
          { extension.enable = lib.mkEnableOption "the bloated dependency"; }
```

If these blocks are markdown, this should be formatted as a code block:

```suggestion
      ```nix
      { extension.enable = lib.mkEnableOption "the bloated dependency"; }
      ```
```

If the language is by default `nix`, we can drop it:

```suggestion
      ```
      { extension.enable = lib.mkEnableOption "the bloated dependency"; }
      ```
```

Unsure whether the doc-comment RFC specifies blocks having a default language.

---

It does not. The default is no highlighting.

Note that if you ever add `nixdoc` generated docs to your website, it will spit out the markdown more or less as-is; so it'll be mdbook rendering it.

As for `nix repl`, currently it doesn't do much (any?) syntax highlighting.

In nixpkgs, it is convention to use ```` ```nix ```` for examples, and ```` ``` ```` for types.